tools/wake-format/emitter.cpp

```diff
@@ -1284,7 +1285,7 @@ wcl::doc Emitter::walk_type(ctx_t ctx, CSTElement node) {
                .nest(fmt().freshline().walk(WALK_NODE).consume_wsnlc())
                .freshline()
                .lit(wcl::doc::lit(")"))
-               .format(ctx, node, token_traits));
+               .compose(ctx, node, token_traits));  // TODO: find a way to make this not-compose
```

Comment:
What is this it terrifies me.

Reply:
Yeah, I don't like it either. Its a similar problem to the "walk_children" problem mentioned before.

Basically, since `walk_type` isn't looking into node (via `node.firstChildElement`) it is possible (and likely) that there is some unconsumed sibling that should be consumed somewhere else. Doing `format` causes an assert

And it isn't acceptable to use `node.firstChildElement` since node is the type we are formatting and we loose the outer apply/paren/whatever if we look at the child.